### PR TITLE
Adds pagination to the patient list

### PR DESCRIPTION
**Pull request by rkodev**
Adds pagination to the patient list.

Closes https://github.com/opensrp/fhircore/issues/54

### CODE REVIEW

File: android/app/src/main/java/org/smartregister/fhircore/fragment/PatientListFragment.kt [hunk 1]
```diff
@@ -74,11 +77,13 @@ class PatientListFragment : Fragment() {
         )
         .get(PatientListViewModel::class.java)
 
-    patientListViewModel.liveSearchedPatients.observe(
+    patientListViewModel.liveSearchedPaginatedPatients.observe(
       requireActivity(),
       {
-        Log.d("PatientListActivity", "Submitting ${it.count()} patient records")
-        adapter.submitList(it)
+        Log.d("PatientListActivity", "Submitting ${it.first.count()} patient records")
```
Comment (Rkareko) [inline android/app/src/main/java/org/smartregister/fhircore/fragment/PatientListFragment.kt:83]: Could we use Timber for logging?

File: android/app/src/main/java/org/smartregister/fhircore/viewmodel/PatientListViewModel.kt [hunk 2]
```diff
@@ -59,7 +60,7 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
 
   val liveSearchPatient: MutableLiveData<PatientItem> by lazy { MutableLiveData<PatientItem>() }
 
-  fun getSearchResults(query: String? = null) {
+  fun getSearchResults(query: String? = null, page: Int = 0, count: Int = 10) {
```
Comment (Rkareko) [inline android/app/src/main/java/org/smartregister/fhircore/viewmodel/PatientListViewModel.kt:63]: Could we rename `count` to somehting more intuitive like `itemCount`

File: android/app/src/main/java/org/smartregister/fhircore/viewmodel/PatientListViewModel.kt [hunk 3]
```diff
@@ -76,17 +77,47 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
...
+        Pair(
+          samplePatients.getPatientItems(searchResults),
+          Pagination(totalItems = getCount(query), pageSize = count, currentPage = page)
+        )
     }
   }
 
+  /** Returns number of records in database */
+  // TODO This is a wasteful query that will be replaced by implementation of
+  // https://github.com/google/android-fhir/issues/458
```
Comment (Rkareko) [inline android/app/src/main/java/org/smartregister/fhircore/viewmodel/PatientListViewModel.kt:94]: Could we create a ticket to track this on this repo

**Review by vincent-karuri — APPROVED** [review]
lgtm

**Pull request merged**